src/webapi/rendering_context.rs

```diff
@@ -27,7 +190,809 @@ impl RenderingContext for CanvasRenderingContext2d {
     }
 }
 
+impl CanvasGradient {
+
+    /// Adds a new stop, defined by an offset and a color, to the gradient. If the offset is 
+    /// not between 0 and 1, an INDEX_SIZE_ERR is returned, if the color can't be parsed as a 
+    /// CSS <color>, a SYNTAX_ERR is returned.
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasGradient/addColorStop)
+    // https://html.spec.whatwg.org/#2dcontext:dom-canvasgradient-addcolorstop
+    pub fn add_color_stop(&self, offset: f64, color: &str) -> Result<(), AddColorStopError> {
+        js_try! ( @(no_return)
+            @{&self.0}.addColorStop(@{offset}, @{color});
+        ).unwrap()
+    }
+}
+
+impl ImageData {
+
+    /// Returns a Uint8ClampedArray representing a one-dimensional array containing the data in the RGBA order, 
+    /// with integer values between 0 and 255 (included).
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/ImageData/data)
+    // https://html.spec.whatwg.org/#2dcontext:dom-imagedata-data
+    // TODO: Return Uint8ClampedArray reference PR 96: https://github.com/koute/stdweb/pull/96 
+    /*
+    pub fn get_data(&self) -> TypedArray<u8> {
+        js! (
+            return @{&self.0}.data;
+        ).try_into().unwrap()
+    }*/
+
+    /// Returns the number of rows in the image data object.
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/ImageData/height)
+    // https://html.spec.whatwg.org/#2dcontext:dom-imagedata-height
+    pub fn get_height(&self) -> u32 {
+        js! (
+            return @{&self.0}.height;
+        ).try_into().unwrap()
+    }
+
+    /// Returns the number of pixels per row in the image data object.
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/ImageData/width)
+    // https://html.spec.whatwg.org/#2dcontext:dom-imagedata-width
+    pub fn get_width(&self) -> u32 {
+        js! (
+            return @{&self.0}.width;
+        ).try_into().unwrap()
+    }
+}
+
 impl CanvasRenderingContext2d {
+    /// # Properties
+    
+    /// The CanvasRenderingContext2D.canvas property is a read-only reference to the HTMLCanvasElement 
+    /// object that is associated with the context. It might be null if there is no association with an <canvas> element.
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/canvas)
+    // https://html.spec.whatwg.org/#2dcontext:dom-context-2d-canvas
+    pub fn get_canvas(&self) -> Option<CanvasElement> {
```

Comment:
I know the MDN says it can return `null`, but in the specs we don't have a `?` after the type, so, are you sure it can return null? (If yes then adding a simple `#[test]` which demonstrates that would be nice.)

Reply:
I wasn't able to produce a null value from accessing the `canvas` property.